src/citrine/_serialization/properties.py

```diff
@@ -134,6 +134,23 @@ def __str__(self):
         return '<Property {!r}>'.format(self.serialization_path)
 
 
+class PropertyCollection(Property[DeserializedType, SerializedType]):
+
+    def __set__(self, obj, value: typing.Union[SerializedType, DeserializedType]):
+        """Property setter, deferring to the setter of the parent class, if applicable."""
+        base_class = _get_base_class(obj, self.serialization_path)
+        if issubclass(type(value), self.underlying_types):
+            value_to_set = self._set_elements(value)
```

Comment:
This is a forward reference to `_set_elements`.  Can you make `PropertyCollection` abstract (or throw NotImplementedErrors) and define an abstract (or NotImplemented) `_set_elements` with a docstring?

Reply:
I went with NotImplemented, but we could switch to abstract if you think it's worth adding that in.

I tend to prefer not using it unless I have to, because it adds some complexity and "dark magic" that I've seen cause issues in the past (I don't think it would cause issues here, but it doesn't seem necessary either)

Reply:
woops, looks like python 3.5 and 3.6 didn't like that, gonna fix it and make them happy

Reply:
Went with @abstractmethod to match existing implementations